cni/linux.Dockerfile

```diff
@@ -0,0 +1,38 @@
+ARG ARCH
+ARG DROPGZ_VERSION=v0.0.12
+ARG OS_VERSION
+ARG OS
+
+FROM mcr.microsoft.com/oss/go/microsoft/golang:1.21 AS azure-vnet
+ARG OS
```

Comment:
your guess is correct. inside each `FROM` is as separate build context. the `ARG` defs in the global scope are required to use the `ARG`s _in_ the `FROM` line (which we do in the Windows build, and I have left here for consistency and because I want to collapse the two Dockerfiles in to a single multiplatform one, next).